tools/protocol-schema-check/src/main.rs

```diff
@@ -0,0 +1,107 @@
+use near_primitives::hash::CryptoHash;
+use near_structs_checker::ProtocolStruct;
+use near_structs_checker_core::ProtocolStructInfo;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::{HashMap, HashSet};
+use std::fs;
+use std::hash::{Hash, Hasher};
+// #[derive(Copy, Clone, Debug, ProtocolStruct)]
+// pub struct CryptoHash(pub [u8; 33]);
+
+fn compute_hash(
+    info: &ProtocolStructInfo,
+    structs: &HashMap<&'static str, &'static ProtocolStructInfo>,
+) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    match info {
+        ProtocolStructInfo::Struct { name, fields } => {
+            name.hash(&mut hasher);
+            for (field_name, field_type) in *fields {
+                field_name.hash(&mut hasher);
```

Comment:
Hm, a conundrum here. In principle the field names do not matter for borsh and so there will be false-positives when somebody e.g. renames a field from `foo` to `foo_deprecated`, but I also don't see how we'd otherwise detect that somebody swapped two adjacent u64 fields (a good reason to always use newtypes!)...

Reply:
Staying with field names for now.